tools/run_tests/xds_test_driver/README.md

```diff
@@ -0,0 +1,74 @@
+# xDS Kubernetes Interop Tests
+
+Proxyless Security Mesh Interop Tests executed on Kubernetes. Work in progress.
```

Comment:
This still says work in progress. (which might be fine, but we should expand on that somewhat).

In general, given the time constraints and the massive size of this PR, any review is likely to miss some items. What I think we need here in the readme are:

1) A clear description of how this interoperates (or does not) with the existing `run_xds_tests.py`, including if there are any plans to deprecate that script and rely on the contents of `xds_test_driver/` exclusively (it _looks_ like much of the functionality is included in this PR, but I'm not clear on if this PR can run non-k8s tests at this point). Coming into this fresh, it would seem that run_xds_tests.py and the xds_test_driver/ directory are related, when currently they are separate things.
2) A plan for how to move this from "experimental/work in progress" to code that is fully-reviewed and complete. Do we want to put this in a `./run_tests/xds_test_driver/alpha/` directory for now to indicate the current status? I'm not sure that's a great approach - what will github's UI show when the files are relocated in a subsequent PR? - so suggestions welcome.

Reply:
My main goal of "work in progress/experimental" wording is to reserve a right to change/restructure this before letting other folks make changes/add features to it. I want to avoid breaking their work.
I wish I had more time to give it a final lift up, but, on the other hand, It's _already_ 4,000 lines and way overdue for a feedback.
README.md was more an afterthought than a proper documentation. Given time constraints, my main effort was on making it work.
Let's discuss short-term approach, and future plans for it over GVC?

Reply:
That makes sense. Happy to discuss over GVC (I know it's late for your timezone/early for me right now, so not sure if that will line up). 

I think putting a note here in the readme that (ideally) lists at least a couple of the concrete things needed (kind of like a high-level set of TODOs) before the "work in progress" note here can be removed would address the first point from my earlier comment. If there's not an easy to produce list (maybe just "address all inline TODOs?"), just something to the effect of what you said in your reply here would be sufficient too. I'll note that my main concern is not that you will need to make changes to this breaking other people's work, but rather that you may inevitably get drawn into other high-priority work and this will remain as-is until something breaks, at which point the structure here will no longer be fresh in anyone's mind.

Regarding distinguishing between this and run_xds_tests.py - how about just renaming this directory to `xds_k8s_test_driver`? Then we can at least easily point to the difference between this PR and the existing run_xds_tests.py setups.